File: Assets/Bundles/Shared/Resources/TS/Signals/BlockPlaceSignal.lua
-- Compiled with unity-ts v2.1.0-75
local GetItemMeta = require("Shared/TS/Item/ItemDefinitions").GetItemMeta
local BlockPlaceSignal
do
	BlockPlaceSignal = setmetatable({}, {
		__tostring = function()
			return "BlockPlaceSignal"
		end,
	})
	BlockPlaceSignal.__index = BlockPlaceSignal
	function BlockPlaceSignal.new(...)
		local self = setmetatable({}, BlockPlaceSignal)
		return self:constructor(...) or self
	end
	function BlockPlaceSignal:constructor(pos, itemType, voxel, entity)
		self.pos = pos
		self.itemType = itemType
		self.voxel = voxel
		self.entity = entity
		self.itemMeta = GetItemMeta(itemType)
	end
end
return {
	BlockPlaceSignal = BlockPlaceSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Shared/Signals/BlockPlaceSignal.ts
import { Entity } from "Shared/Entity/Entity";
import { GetItemMeta } from "Shared/Item/ItemDefinitions";
import { ItemMeta } from "Shared/Item/ItemMeta";
import { ItemType } from "Shared/Item/ItemType";

export class BlockPlaceSignal {
	public readonly itemMeta: ItemMeta;

	constructor(
		public readonly pos: Vector3,
		public readonly itemType: ItemType,
		public readonly voxel: number,

		/** Will always be undefined on client. */
		public readonly entity?: Entity,
	) {
		this.itemMeta = GetItemMeta(itemType);
	}
}
